frontend/packages/dev-console/src/utils/application-utils.ts

```diff
@@ -414,6 +409,7 @@ export const cleanUpWorkload = (
   const batchDeleteRequests = (models: K8sKind[], resourceObj: K8sResourceKind): void => {
     models.forEach((model) => deleteRequest(model, resourceObj));
   };
+  if (isKnativeEventResource) deleteRequest(modelFor(referenceFor(resource)), resource);
```

Comment:
can we have `if (isDynamicEventResourceKind(referenceFor(resource)))` ?

Reply:
fixed!